Makefile

```diff
@@ -270,7 +274,10 @@ deploy-onprem-for-subsystem:
 	export DUMMY_IGNITION="true" && $(MAKE) deploy-onprem
 
 deploy-on-openshift-ci:
-	echo "placeholder for deployment on openshift ci"
+	ln -s $(shell which oc) $(shell dirname $(shell which oc))/kubectl
```

Comment:
Why?

Reply:
The current build process for subsystem is relying on the kubectl command. However, kubectl cli is not provided by openshift ci. In order to avoid either fetching kubectl or changing the current logic, we instead link kubectl to oc, so that we can still use the same command as before, but on openshift ci kubectl will call oc under the hood.

Reply:
Don't we run inside a build container that we can set whatever executable we want?
That's what `Dockerfile.assisted-service-build` is for.

Reply:
We get `oc` on `test-bin` image as part of openshift ci platform. To get `kubectl` on `test-bin`, we will have to download it from somewhere on the fly. Alternatively, we can modify the existing subsystem procedures to also support `oc` (so that jenkins can still use `kubectl`, while openshift ci uses `oc`).
The alternative we came up with is to give `oc` another name on `test-bin` - `kubectl`. This way, both openshift ci and jenkins procedure can use `kubectl`, but under the hood, a different binary will be triggered.

`Dockerfile.assisted-service-build` is the dockerfile for one of the images `test-bin` instantiates. We need `kubectl` available on `test-bin` rather than on `assisted-service`

Reply:
Sounds a little bit hacky. Is `oc` mandatory for handling on top of OpenShift CI?

Reply:
`oc` is not mandatory for openshift ci, but openshift ci provides a simple mechanism to make it available (we set `cli: latest` under `test` and it does the rest).
Having said that, `kubectl` IS required for assisted-service CI, so we need to provide it somehow.
I think the solution we propose is not that hacky, as `oc` knows how to return `kubctl`-like outputs when it's called via the linked `kubectl` command - I'll add an example below.
The alternatives of either maintaining code fetching the `kubectl` binaries, or refactoring the CI processes to support both `kubectl` and `oc` seems more "risky" to me.
My perspective is that this hack is so neat in the sense that it does not require any changes beyond that line, so it would be a shame not to use it, at least until we think it raises any concern.

```
[tester@yherzog-dev1 ~]$ ll ~/.crc/bin/oc/kubectl
lrwxrwxrwx. 1 tester tester 27 Dec  3 15:07 /home/tester/.crc/bin/oc/kubectl -> /home/tester/.crc/bin/oc/oc
[tester@yherzog-dev1 ~]$ oc version
Client Version: 4.6.1
Server Version: 4.6.1
Kubernetes Version: v1.19.0+d59ce34
[tester@yherzog-dev1 ~]$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"074039a0a9c137967fba3e667b9849d60e5054d8", GitTreeState:"clean", BuildDate:"2020-10-08T07:17:20Z", GoVersion:"go1.15.0", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0+d59ce34", GitCommit:"d59ce3486ae3ca3a0c36e5498e56f51594076596", GitTreeState:"clean", BuildDate:"2020-10-08T15:58:07Z", GoVersion:"go1.15.0", Compiler:"gc", Platform:"linux/amd64"}
```